lib/god/socket.rb

```diff
@@ -95,8 +95,14 @@ def start
       end
 
       if File.exists?(self.socket_file)
-        uid = Etc.getpwnam(@user).uid if @user
-        gid = Etc.getgrnam(@group).gid if @group
+        if @user
+          user_method = @user.is_a?(Integer) ? :getpwuid : :getpwnam
+          uid = Etc.send(user_method, @user).uid
```

Comment:
It seems like this should also do the same gid defaulting logic from the `Process` class too.